rational/src/third_party/ser.rs

```diff
@@ -0,0 +1,43 @@
+use serde::ser::SerializeStruct;
+use serde::{Serialize, Serializer};
+
+use dashu_base::Sign;
+
+use crate::RBig;
+
+impl Serialize for RBig {
+    fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        if serializer.is_human_readable() {
+            serializer.serialize_str(&self.to_string())
+        } else {
+            let sign = self.sign();
+            let numerator = self.numerator().clone().into_parts().1.to_le_bytes();
```

Comment:
There is one more clone here, because the binary form cannot be obtained directly